### PR TITLE
Azurefilesystem update

### PR DESCRIPTION
**Pull request by plutoshe**
update the code, deps of azure filesystem

### CODE REVIEW

**Comment by BaiGang** [comment]
LGTM.

File: .script/dep.sh [hunk 1]
```diff
@@ -8,4 +8,8 @@ go get -u github.com/colinmarc/hdfs
 go get -u golang.org/x/net/context
 go get -u google.golang.org/grpc
 go get -u github.com/golang/protobuf/proto
-go get -u github.com/MSOpenTech/azure-sdk-for-go/storage
+go get -u golang.org/x/tools/cmd/vet
```
Comment (fengjingchao) [inline .script/dep.sh]: We don't need these two deps, right?

Comment (plutoshe) [inline .script/dep.sh]: But azure need these two deps.

Comment (fengjingchao) [inline .script/dep.sh]: These script is used to track high level deps.
Those sub-deps will be downloaded automatically.

Comment (plutoshe) [inline .script/dep.sh]: But I am confused that travis build failed if these two deps don't exist. 
Thus I follow the instruction of <a href = "https://github.com/Azure/azure-sdk-for-go/commit/383dafbdf92619345cb75e933524504333193f36"> this </a>

Comment (fengjingchao) [inline .script/dep.sh]: Try it.
These two deps shouldn't show in the script because they aren't used.

**Comment by plutoshe** [comment]
Seems working well this time

**Comment by fengjingchao** [comment]
LGTM

**Pull request merged**